packages/web-app/components/create/AgreementCreationForm.tsx

```diff
@@ -16,6 +16,20 @@ export const AgreementCreationForm = () => {
 		return validateCriteria(positions);
 	}, [terms, positions]);
 
+	const [participantRow, setParticipantRow] = useState([ParticipantRow]);
+
+	// handle click event to add button.
+	const handleAddClick = () => {
+		setParticipantRow([...participantRow, ParticipantRow]);
+	}
+
+	// handle click event to remove button.
+	const handleRemoveClick = (index: number) => {
+		const list = [...participantRow];
+		list.splice(index, 1);
+		setParticipantRow(list);
+	};
+
```

Comment:
You should be updating positions, not creating a new array of react elements. You are not using this array for more than the length. When you add more than two elements here you create a mismatch between the length of array of positions that is used in PaticipantRow components and the number of ParticipantRow components your are rendering.